File: src/main/java/leet_code/till_april/LongestCommonPrefix.java
package leet_code.till_april;

public class LongestCommonPrefix {


//    Write a function to find the longest common prefix string amongst an array of strings.
//
//    If there is no common prefix, return an empty string "".
//
//
//
//    Example 1:
//
//    Input: strs = ["flower","flow","flight"]
//    Output: "fl"
//    Example 2:
//
//    Input: strs = ["dog","racecar","car"]
//    Output: ""
//    Explanation: There is no common prefix among the input strings.


    public static void main(String[] args) {
        System.out.println(longestCommonPrefix(new String[]{"lily", "lil", "lilbil"}));
//        System.out.println("lily".indexOf("il"));

    }


    public static String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) return "";
        String prefix = strs[0];
        for (int i = 1; i < strs.length; i++) {

            while (strs[i].indexOf(prefix) != 0) {
                prefix = prefix.substring(0, prefix.length() - 1);
            }

        }
        return prefix;
    }
}

File: src/main/java/leet_code/top_interview_questions_easy/string/LongestCommonPrefix.go
package main

import "fmt"

func main() {
	fmt.Println(longestCommonPrefix([]string{"flower", "flow", "flight"}))
	fmt.Println(longestCommonPrefix([]string{"flower", "flow", "floght"}))
}

func longestCommonPrefix(strs []string) string {

	if len(strs) == 0 {
		return ""
	}

	prefix := strs[0]
	for i := 1; i < len(strs); i++ {
		currentWord := strs[i]
		j := 0
		for j < len(prefix) && j < len(currentWord) && prefix[j] == currentWord[j] {
			j++
		}
		if j == 0 {
			return ""
		}
		prefix = prefix[:j]
	}
	return prefix
}
